### PR TITLE
Cu 86bx0nhx7/be get user profile details

### PR DESCRIPTION
**Pull request by Don-christo**
**Description of PR**

To get a user's profile details.

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [x] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bx0nhx7 BE-get User profile details](https://app.clickup.com/t/86bx0nhx7)

File: apps/server/src/controllers/userControllers/getUserProfileDetails.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        gender: user.gender,
+        occupation: user.occupation,
+        date_of_birth: user.date_of_birth,
+        bvn: user.bvn,
+        address: user.address,
+        identification_type: user.identification_type,
+        identification_number: user.identification_number,
+        identification_doc: user.identification_doc,
+        proof_of_address_doc: user.proof_of_address_doc,
+        isVerified: user.isVerified,
```
Comment (polymathmode) [inline apps/server/src/controllers/userControllers/getUserProfileDetails.ts:24]: I'm wondering if its very necessary to return this entire object??

File: apps/server/src/controllers/userControllers/userUpdateKyc.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      identification_type,
+      identification_number,
+      identification_doc,
+      proof_of_address_doc,
+    } = validationResult.data
+
+    const files = req.files as {
+      [fieldname: string]: Express.Multer.File[]
+    }
+    console.log(files)
```
Comment (polymathmode) [inline apps/server/src/controllers/userControllers/userUpdateKyc.ts:39]: You can also remove the logged files

**Review by polymathmode — COMMENTED** [review]
Well done!

**Review by pelzman — COMMENTED** [review]
image
it would be better if the pagination could be pass as query string, rather than as object, if its possible

File: apps/server/src/controllers/userControllers/getUserProfileDetails.ts (hunk 1 above)
Comment (Don-christo) [reply apps/server/src/controllers/userControllers/getUserProfileDetails.ts:24]: Well, what is needed are the profile details, which are also the user object. Shouldn't I return the entire object??

File: apps/server/src/controllers/userControllers/userUpdateKyc.ts (hunk 2 above)
Comment (Don-christo) [reply apps/server/src/controllers/userControllers/userUpdateKyc.ts:39]: I thought I've removed this already. Okay, will do so.

Comment (Don-christo) [reply apps/server/src/controllers/userControllers/userUpdateKyc.ts:39]: Oh, now I remember. I've already done that in the updateUserKyc branch. I also refactored some few codes in the branch. So I think when they are all merged, it'll be fixed.

**Comment by KristofaJosh** [comment]
I can see an outdated code here (not sure, seems like something that have been corrected in one of your PRs) that if merged will overwrite the good code.

so make sure to always start your Task on clean branch

**Pull request merged**